homeassistant/components/date/__init__.py

```diff
@@ -0,0 +1,132 @@
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
+__all__ = ["DOMAIN", "DateEntity", "DateEntityDescription"]
+
+
+async def _async_set_value(entity: DateEntity, service_call: ServiceCall) -> None:
+    """Service call wrapper to set a new date."""
+    return await entity.async_set_value(service_call.data[ATTR_DATE])
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Date entities."""
+    component = hass.data[DOMAIN] = EntityComponent[DateEntity](
+        _LOGGER, DOMAIN, hass, SCAN_INTERVAL
+    )
+    await component.async_setup(config)
+
+    component.async_register_entity_service(
+        SERVICE_SET_VALUE, {vol.Required(ATTR_DATE): cv.date}, _async_set_value
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent[DateEntity] = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent[DateEntity] = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+@dataclass
+class DateEntityDescription(EntityDescription):
+    """A class that describes date entities."""
+
+
+class DateEntity(Entity):
+    """Representation of a Date entity."""
+
+    entity_description: DateEntityDescription
+    _attr_native_value: date | None
+
+    @property
+    @final
+    def state_attributes(self) -> dict[str, int]:
+        """Return the state attributes."""
+        state_attr: dict[str, int | None] = {
+            ATTR_DAY: self.day,
+            ATTR_MONTH: self.month,
+            ATTR_YEAR: self.year,
+        }
+        return {k: v for k, v in state_attr.items() if v is not None}
+
+    @property
+    @final
+    def state(self) -> str | None:
+        """Return the entity state."""
+        if self.native_value is None:
+            return None
+        return self.native_value.strftime(FORMAT_DATE)
+
+    @property
+    @final
+    def day(self) -> int | None:
+        """Return day from value."""
+        if self.native_value is None:
+            return None
+        return self.native_value.day
+
+    @property
+    @final
+    def month(self) -> int | None:
+        """Return month from value."""
+        if self.native_value is None:
+            return None
+        return self.native_value.month
+
+    @property
+    @final
+    def year(self) -> int | None:
+        """Return year from value."""
+        if self.native_value is None:
+            return None
+        return self.native_value.year
+
+    @property
+    def native_value(self) -> date | None:
+        """Return the value reported by the date."""
+        return self._attr_native_value
+
+    def set_value(self, date_value: date) -> None:
```

Comment:
Just a tiny suggestion to make it a little more clean/mean

```suggestion
    def set_value(self, value: date) -> None:
```